x-pack/plugins/security_solution/cypress/integration/overview/cti_link_panel.spec.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  OVERVIEW_CTI_ENABLE_MODULE_BUTTON,
+  OVERVIEW_CTI_LINKS,
+  OVERVIEW_CTI_LINKS_ERROR_INNER_PANEL,
+  OVERVIEW_CTI_LINKS_INFO_INNER_PANEL,
+  OVERVIEW_CTI_LINKS_WARNING_INNER_PANEL,
+  OVERVIEW_CTI_TOTAL_EVENT_COUNT,
+  OVERVIEW_CTI_VIEW_DASHBOARD_BUTTON,
+} from '../../screens/overview';
+
+import { loginAndWaitForPage } from '../../tasks/login';
+import { OVERVIEW_URL } from '../../urls/navigation';
+import { cleanKibana } from '../../tasks/common';
+import { esArchiverLoad, esArchiverUnload } from '../../tasks/es_archiver';
+
+describe('CTI Link Panel', () => {
+  before(() => {
+    cleanKibana();
+  });
+
+  describe('disabled threat intel module', () => {
+    it('renders error inner panel', () => {
+      loginAndWaitForPage(OVERVIEW_URL);
+      cy.get(`${OVERVIEW_CTI_LINKS} ${OVERVIEW_CTI_LINKS_ERROR_INNER_PANEL}`).should('exist');
+    });
+
+    it('renders disabled "view dashboard" button', () => {
+      cy.get(`${OVERVIEW_CTI_VIEW_DASHBOARD_BUTTON}`).should('be.disabled');
+    });
+
+    it('renders 0 total event count', () => {
+      cy.get(`${OVERVIEW_CTI_TOTAL_EVENT_COUNT}`).should('have.text', 'Showing: 0 events');
+    });
+
+    it('renders enable module button', () => {
+      it('renders disabled "view dashboard" button', () => {
+        cy.get(`${OVERVIEW_CTI_ENABLE_MODULE_BUTTON}`).should('exist');
+        // TODO: how to test injected doclink?
+        // cy.get(`${OVERVIEW_CTI_ENABLE_MODULE_BUTTON} a`).should('have.attr', 'href', "doclink");
+      });
+    });
+  });
+
+  describe('enabled threat intel module', () => {
+    before(() => {
+      esArchiverLoad('threat_indicator');
+    });
+
+    after(() => {
+      esArchiverUnload('threat_indicator');
+    });
+
+    describe('disabled dashboard module', () => {
+      describe('when there are no events in the selected time period', () => {
```

Comment:
Broadly: this is a lot of permutations to exercise in an end-to-end test like this. I would recommend focusing on "happy paths" primarily for this level, and relegating most edge cases to jest or possibly FTR. If it's a particularly hairy edge case that does involve a lot of dependencies, that makes sense here, but I don't think we should be testing error/warning/"no data" states at this level.

Reply:
updated test format a bit, let me know if it looks better

Reply:
IMO there's still too many edge cases being tested in an end-to-end test, but this should be less burdensome than before 👍